src/App.tsx

```diff
@@ -1,7 +1,44 @@
-import React from "react";
+import React, { useState, createContext } from "react";
+import { BrowserRouter, Routes, Route } from "react-router-dom";
+import Navbar from "./Components/Navbar";
+import Home from "./Pages/Home";
+import About from "./Pages/About";
+import Contact from "./Pages/Contact";
+import Modal from "./Components/Modal";
+
+type ModalType = {
+  open: string;
+  openModal: (name: string) => void;
+  closeModal: () => void;
+};
+
+const ModalContext = createContext<ModalType>({} as ModalType);
 
 function App() {
-  return <h1>Sidebar-modal</h1>;
+  const [open, setOpen] = useState<string>("");
+
+  const openModal = (name: string) => {
+    setOpen(name);
+  };
+
+  const closeModal = () => {
+    setOpen("");
+  };
```

Comment:
lets use custom hook for this
will sync on this today

Reply:
created a new useModal hook